src/app/features/tech-record/components/change-vehicle-type/change-vehicle-type.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+import { Location } from '@angular/common';
+import { GlobalErrorService } from '@core/components/global-error/global-error.service';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { CustomFormGroup, FormNode, FormNodeTypes } from '@forms/services/dynamic-form.types';
+import { StatusCodes, TechRecordModel, VehicleTechRecordModel, VehicleTypes } from '@models/vehicle-tech-record.model';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { take } from 'rxjs';
+import { getOptionsFromEnum, getOptionsFromEnumAcronym } from '@forms/utils/enum-map';
+import { MultiOptions } from '@forms/models/options.model';
+import { changeVehicleType } from '@store/technical-records';
+import { Store } from '@ngrx/store';
+import { TechnicalRecordServiceState } from '@store/technical-records/reducers/technical-record-service.reducer';
+import { ActivatedRoute, Router } from '@angular/router';
+
+@Component({
+  selector: 'app-change-vehicle-type',
+  templateUrl: './change-vehicle-type.component.html'
+})
+export class ChangeVehicleTypeComponent implements OnInit, OnChanges {
+  constructor(
+    private technicalRecordService: TechnicalRecordService,
+    public globalErrorService: GlobalErrorService,
+    private store: Store<TechnicalRecordServiceState>,
+    private router: Router,
+    private route: ActivatedRoute,
+    public dfs: DynamicFormService
+  ) {
+    this.technicalRecordService.selectedVehicleTechRecord$.pipe(take(1)).subscribe(data => (this.vehicleTechRecord = data));
+    this.technicalRecordService.editableTechRecord$.pipe(take(1)).subscribe(data => (this.currentTechRecord = data));
+    this.form = this.dfs.createForm(this.template) as CustomFormGroup;
+  }
+
+  public currentTechRecord?: TechRecordModel;
+  public vehicleTechRecord?: VehicleTechRecordModel;
+  public form!: CustomFormGroup;
+
+  public template: FormNode = {
+    name: 'criteria',
```

Comment:
would be nice to have some unit tests on this component

Reply:
I do need to learn how to write them properly though.. my method is hack at random until the number goes up and the lights turn green..
If anyone has any ... err ... specific priorities ... I'd like to know them